### PR TITLE
Add X-Request-Id logging

### PR DESCRIPTION
**Pull request by MHova**

### CODE REVIEW

File: src/Canteven/HTTP.hs [hunk 1]
```diff
@@ -150,6 +152,10 @@ requestLogging :: LoggerTImpl -> Middleware
 requestLogging logging app req respond = (`runLoggingT` logging) $ do
     $(logInfo) . pack
       $ "Starting request: " ++ reqStr
+    $(logInfo) $ maybe
+      "X-Request-Id header missing"
+      ((T.append) "X-Request-Id: " . decodeUtf8)
```
Comment (potomak) [inline src/Canteven/HTTP.hs]: Do you need () around append?

File: src/Canteven/HTTP.hs [hunk 2]
```diff
@@ -24,12 +24,14 @@ import Control.Monad (void, join)
...
 import Control.Monad.Trans.Class (lift)
-import Data.List ((\\))
+import Data.ByteString (ByteString)
+import Data.List ((\\), find)
 import Data.Maybe (catMaybes)
 import Data.Monoid ((<>))
 import Data.Text (Text, pack, unpack)
 import Data.Text.Encoding (encodeUtf8, decodeUtf8)
 import Data.Time.Clock (getCurrentTime, UTCTime, diffUTCTime)
+import Data.Tuple (fst, snd)
```
Comment (potomak) [inline src/Canteven/HTTP.hs]: Those should be included in Prelude

File: src/Canteven/HTTP.hs (hunk 1 above)
Comment (MHova) [inline src/Canteven/HTTP.hs]: Fixed

File: src/Canteven/HTTP.hs (hunk 2 above)
Comment (MHova) [inline src/Canteven/HTTP.hs]: Fixed

File: src/Canteven/HTTP.hs [hunk 3]
```diff
@@ -150,6 +151,10 @@ requestLogging :: LoggerTImpl -> Middleware
 requestLogging logging app req respond = (`runLoggingT` logging) $ do
     $(logInfo) . pack
       $ "Starting request: " ++ reqStr
+    $(logInfo) $ maybe
+      "X-Request-Id header missing"
+      (T.append "X-Request-Id: " . decodeUtf8)
+      requestHeaderRequestId
```
Comment (taphu) [inline src/Canteven/HTTP.hs]: Why not make this part of the "Starting request:" log message? Like "Starting request with id xxx" or "Starting request with no request id"

**Pull request merged**